src/mpi/comm/comm_split_type.c

```diff
@@ -154,6 +154,8 @@ static int node_split_processor(MPIR_Comm * comm_ptr, int key, hwloc_obj_type_t
 
     obj_containing_cpuset =
         hwloc_get_obj_covering_cpuset(MPIR_Process.topology, MPIR_Process.bindset);
+    if (obj_containing_cpuset == NULL)
+        goto fn_exit;
```

Comment:
If these processes skip `MPIR_Comm_split_impl` won't it cause a deadlock?

Reply:
yes, that is possible if some processes have a valid bind set while some end up with  obj_containing_cpuset. I'll update the patch with a fix for this. Thank you.